classes/manager.php

```diff
@@ -0,0 +1,101 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * This file contains functions to get various observation data objects
+ *
+ * @package   mod_observation
+ * @copyright  2021 Endurer Solutions Team
+ * @author Matthew Hilton <mj.hilton@outlook.com>
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace mod_observation;
+
+defined('MOODLE_INTERNAL') || die();
+
+/**
+ * mod_observation management class
+ *
+ * @package   mod_observation
+ * @copyright  2021 Endurer Solutions Team
+ * @author Matthew Hilton <mj.hilton@outlook.com>
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class manager {
+    /**
+     * Gets observation, course and coursemodule from course module ID
+     * @param int $cmid Course module ID
+     * @param string $tablename database table name
+     * @return list List containing the observation instance, course and coursemodule (in that order)
+     */
+    public static function get_observation_course_cm_from_cmid(int $cmid, string $tablename='observation') {
+        global $DB;
+        list($course, $cm) = get_course_and_cm_from_cmid($cmid, $tablename);
+        $observationid = $cm->instance;
+        if (!$observation = $DB->get_record($tablename, array('id' => $observationid))) {
+            throw new moodle_exception('moduleinstancedoesnotexist');
+        }
+        return array($observation, $course, $cm);
```

Comment:
A decent habit to get into is to shorten array calls to []. Eg array('id' => $obsid) becomes ['id' => $obsid]

Reply:
Im not fussed either way, the most important thing is consistency, but is more clean + concise